### DISCUSSION TITLE
RuntimeError: No algebra backend available!

### DISCUSSION THREAD

**Discussion by Ashdore** [osqp/osqp#657] [Python Interface] [answered]
I am using OSQP in my code on Python 3.12.4 with Numpy 2.0.0. OSQP has been build on my computer using Numpy 2.0.0. My code runs fine in Python. When I create an .exe file using PyInstaller (single file or not), I get this error:

  File "osqp\interface.py", line 80, in __init__
  File "osqp\interface.py", line 49, in default_algebra
RuntimeError: No algebra backend available!
[13168] Failed to execute script 'lineid' due to unhandled exception!

Any solution? Thanks

**Comment by Ashdore** [comment, marked as answer]
Module `osqp.ext_builtin`, `osqp_cuda` or `osqp_mkl` should be added to PyInstaller as a hidden import in the command line, e.g.:

`--hidden-import=osqp.ext_builtin`

**Comment by imciner2** [comment]
Please use the branch v0.6.x for OSQP if you are cloning/building yourself (https://github.com/osqp/osqp-python/tree/v0.6.x). The master branch is currently a beta/unstable version based on the new v1.0 C code, so it is not meant for general use.

We are working on getting new wheels built and pushed to PyPi that will support NumPy 2.0, and hope to have that done in a day or so.

That said, I don't think we have ever tried with the future 1.0 on Pyinstaller before (and I have actually never heard of it before). Is there something we can do in our packaging to make this usecase better or work with no problems?

**Reply by Ashdore** [reply]
Thanks for the info. I tried to clone and build v0.6.x but I got this error:

git clone --recurse-submodules https://github.com/osqp/osqp-python/tree/v0.6.x
Cloning into 'v0.6.x'...
fatal: repository 'https://github.com/osqp/osqp-python/tree/v0.6.x/' not found

Am I missing something?

Regarding PyInstaller, my problem was that the backend algebra was not imported for some reason, and I had to hidden import it in the command line:

`--hidden-import=osqp.ext_builtin`

There was another issue with importing `numpy.f2py` that is being take care of by PyInstaller developers.

**Reply by imciner2** [reply]
It is a branch in the repository, so you need to specify the branch when cloning:

```
git clone --recurse-submodules --branch v0.6.x https://github.com/osqp/osqp-python.git
```

**Reply by Ashdore** [reply]
Thank you.
